src/ZoneServer/World/Actors/CombatEntities/Components/BuffComponent.cs

```diff
@@ -157,7 +157,7 @@ public bool Remove(BuffId buffId)
 		/// <returns></returns>
 		public BuffId RemoveRandomBuff()
 		{
-			var removableBuffs = this.GetAll(a => a.Data.Type == BuffType.Buff && a.Data.Removable);
+			var removableBuffs = this.GetAll(a => a.Data.Type == BuffType.Buff && a.Data.RemoveBySkill);
```

Comment:
It based on assumption:
---------------
Removable => could be removable by clicking on the buff
Removable By Skill => could be removable by a skill
---------------

Let me know if this makes no sense so I can revert it.

Reply:
The concern wasn't whether the change potentially made logical sense, but whether it's correct and how it might affect calls to this method that might've assumed it to check `Removable`. I suppose we currently only use these methods from skills, but they're presented as general purpose methods, and changing them to only really be useable for skills could've otherwise broken compatibility. Hence my question whether you checked if this is currently correct and whether you considered this potential issue. Because it should ideally at least be mentioned in the doc comments if this is the new and only intended use.

Reply:
Yeah, it was wrong:

{ id: 1065, className: "LatentVenom_Debuff", name: "Latent Venom", type: "Debuff", tags: ["Poison", "IgnoreDiscernEvil"], level: 99, duration: 100000, updateTime: 1000, overBuff: 1, buffExpUp: 0, removable: false, removeOnDeath: true, removeBySkill: true, save: false, updateProperties: ["ALL"] },


This debuff has removable=false and removableBySkill=true

The same happen to other debuffs:

{ id: 88, className: "Virus_Debuff", name: "Infection", type: "Debuff", tags: ["Poison", "IgnoreImmune"], level: 99, duration: 5000, updateTime: 1000, overBuff: 1, buffExpUp: 0, removable: false, removeOnDeath: true, removeBySkill: true, save: false, updateProperties: ["ALL"] },
{ id: 152, className: "Sleep_Debuff", name: "Sleep", type: "Debuff", tags: ["Sleep"], level: 1, duration: 0, updateTime: 0, overBuff: 1, buffExpUp: 0, removable: false, removeOnDeath: true, removeBySkill: true, save: false, updateProperties: ["ALL"] },